src/resizing.jl

```diff
@@ -311,9 +317,9 @@ imresize_type(c::Gray) = Gray{imresize_type(gray(c))}
 imresize_type(c::FixedPoint) = typeof(c)
 imresize_type(c) = typeof((c*1)/1)
 
-function imresize!(resized::AbstractArray{T,N}, original::AbstractArray{S,N}) where {T,S,N}
+function imresize!(resized::AbstractArray{T,N}, original::AbstractArray{S,N}; method::Interpolations.InterpolationType=BSpline(Linear()), kwargs...) where {T,S,N}
     # FIXME: avoid allocation for interpolation
-    itp = interpolate(original, BSpline(Linear()))
+    itp = interpolate(original, method)
```

Comment:
So here we need a helper function:

```julia
@inline maybe_wrap_with_BSpline(itp::Interpolations.InterpolationType) = itp
@inline maybe_wrap_with_BSpline(degree::Interpolations.Degree) = BSpline(degree)
```

to make sure `imresize!` correctly understands `Linear()`

```suggestion
    itp = interpolate(original, maybe_wrap_with_BSpline(method))
```

Reply:
I see, I was not considering that a case but I can certainly add that in.

Reply:
Since I have specified that the method be ``Interpoaltions.InterpolationType`` I think that fails because ``Interpolations.Linear<:Interpolations.InterpolationType`` fails since it is ``Degree``

Reply:
Nevermind, I just unioned the type.